crates/forge/src/shared_cache.rs

```diff
@@ -0,0 +1,61 @@
+pub const CACHE_DIR: &str = ".snfoundry_cache";
+pub const PREV_TESTS_FAILED: &str = ".prev_tests_failed";
+
+use anyhow::{Ok, Result};
+use camino::Utf8PathBuf;
+use forge_runner::test_case_summary::TestCaseSummary;
+use scarb_metadata::MetadataCommand;
+use std::fs::{self, File};
+use std::io::{BufRead, BufReader, BufWriter, Write};
+
+pub fn get_cached_failed_tests_names(cache_dir_path: &Utf8PathBuf) -> Result<Option<Vec<String>>> {
```

Comment:
Nit, but we could do without `get` in function names

```suggestion
pub fn cached_failed_tests_names(cache_dir_path: &Utf8PathBuf) -> Result<Option<Vec<String>>> {
```

Reply:
IMO, in the context of cache the best way is to use prefix set/get so it could be `get_cached_failed_tests_names`
and setter should be named `set_cached_failed_tests_names`